libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -47,34 +46,62 @@ bool NavEKF3_core::setup_core(NavEKF3 *_frontend, uint8_t _imu_index, uint8_t _c
     _ahrs = frontend->_ahrs;
 
     /*
-      the imu_buffer_length needs to cope with a 260ms delay at a
-      maximum fusion rate of 100Hz. Non-imu data coming in at faster
-      than 100Hz is downsampled. For 50Hz main loop rate we need a
-      shorter buffer.
+      The imu_buffer_length needs to cope with the worst case sensor delay at the
+      maximum fusion rate of 100Hz. Non-imu data coming in at faster than 100Hz is
+      downsampled. For 50Hz main loop rate we need a shorter buffer.
      */
-    if (_ahrs->get_ins().get_sample_rate() < 100) {
-        imu_buffer_length = 13;
+
+    // Calculate the expected EKF time step
+    if (_ahrs->get_ins().get_sample_rate() > 0) {
+        dtEkfAvg = 1.0f / _ahrs->get_ins().get_sample_rate();
+        dtEkfAvg = fmaxf(dtEkfAvg,EKF_TARGET_DT);
```

Comment:
I went through the EKF3 code and both fmaxf() and MAX() functions are being used for floating point constraints in different places. I want to be consistent, but before changing them all to one or the other, would like to know what the advantages and disadvantage are.